### PR TITLE
[Feature] Nesterov momentum optimizer

### PR DESCRIPTION
**Pull request by Lucaspbordignon**
Added the Nesterov momentum optimizer. Furthermore, the README.md file has been updated.

### CODE REVIEW

**Comment by edgarriba** [comment]
@Lucaspbordignon thanks for the contribution ! could you provide tests for this ?

**Comment by Lucaspbordignon** [comment]
@edgarriba yes, of course! I'm not so familiar with CI so I don't know exactly where should I start. Haven't found any test for optimizers, may I create a new file just for this feature?

**Comment by edgarriba** [comment]
right. Please, create a file under the name of `test_optimizers.h`

File: test/test_optimizers.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "gtest/gtest.h"
+#include "tiny_dnn/tiny_dnn.h"
+
+using namespace tiny_dnn::activation;
```
Comment (edgarriba) [inline test/test_optimizers.h]: remove this line

File: tiny_dnn/optimizers/optimizer.h [hunk 2]
```diff
@@ -177,4 +177,31 @@ struct momentum : public stateful_optimizer<1> {
...
+ * A method for unconstrained convex minimization problem with the rate of 
+ * convergence o(1/k2), Doklady ANSSSR, vol.269, pp.543-547, 1983.
+ **/
+struct nesterov_momentum : public stateful_optimizer<1> {
+ public:
+  nesterov_momentum() : alpha(float_t(0.01)), lambda(float_t(0)),
+                        mu(float_t(0.9)) {}
+
+  void update(const vec_t &dW, vec_t &W, bool parallelize) {
+    vec_t &dWprev = get<0>(W);
```
Comment (edgarriba) [inline tiny_dnn/optimizers/optimizer.h]: to disambiguate, `std::get`

File: test/test_optimizers.h (hunk 1 above)
Comment (Lucaspbordignon) [inline test/test_optimizers.h]: Sorry. Was from an old solution. Removed on the latest commit.

File: tiny_dnn/optimizers/optimizer.h (hunk 2 above)
Comment (Lucaspbordignon) [inline tiny_dnn/optimizers/optimizer.h]: Was unable to build when using `std::get`

Comment (edgarriba) [inline tiny_dnn/optimizers/optimizer.h]: you might need to include the header `#include <tuple>`

Comment (edgarriba) [inline tiny_dnn/optimizers/optimizer.h]: try again, otherwise I'll figure out later

Comment (Lucaspbordignon) [inline tiny_dnn/optimizers/optimizer.h]: Still not able to build, even with the `#include <tuple>`

**Comment by edgarriba** [comment]
@Lucaspbordignon thanks ! Do you think you can contribute with tests for the other optimizers ?

**Comment by Lucaspbordignon** [comment]
@edgarriba Sure! Should I open a new PR?

**Pull request merged**